journal/views.py

```diff
@@ -86,20 +86,41 @@ def subscriber_card_view(request, card):
 
     return render(request, template, context)
 
-
-def search(request):
-    result = None
-
+def search_view(request):
+    context = {}
+    template = 'journal/search.html'
     search_input = request.GET.get('search_input', None)
 
-    if search_input is not None and re.match('\d+', search_input):
-        get_object_or_404(PBXPort, subscriber_number=int(search_input))
-        result = redirect(reverse('journal:subscriber_card', args=(search_input,)))
+    if search_input is not None:
+        if re.match('\d+', search_input):
+            pbxport_set = PBXPort.objects.filter(subscriber_number = search_input)
+            number_crosspath = [
+                loads(pbxport.json_path, cls=CrosspathPointDecoder) for pbxport in pbxport_set
+            ]
+            context['number_crosspath'] = number_crosspath
+
+            phone_set = Phone.objects.filter(location__room__room__iregex = r'^[^0-9]*{0}[^0-9]*$'.format(search_input))
```

Comment:
Можно заменить на
```suggestion
            phone_set = Phone.objects.filter(location__room__room__contains=search_input)
```

Reply:
Тогда при вводе "30" будут находиться комнаты 301, 302 303 и т.п.
а если ввести 1 то количество результатов будет зашкаливать

Reply:
Да, точно. Тогда лучше написать такой регэксп: `r'^.*\b{}\b.*$'`
И сделать так:
```suggestion
            room_number_pattern = r'^.*\b{}\b.*$'.format(search_input)
            phone_set = Phone.objects.filter(location__room__room__iregex=room_number_pattern)
```